src/target/riscv/batch.h

```diff
@@ -73,6 +97,25 @@ static inline void riscv_scan_set_delay(struct riscv_scan_delays *delays,
 		delays->sb_write_delay = delay;
 		return;
 	}
+	assert(0);
+}
+
+static inline int riscv_scan_increase_delay(struct riscv_scan_delays *delays,
+		enum riscv_scan_delay_class delay_class)
+{
+	const unsigned int delay = riscv_scan_get_delay(delays, delay_class);
+	const unsigned int delay_step = delay / 10 + 1;
+	if (delay > RISCV_SCAN_DELAY_MAX - delay_step) {
+		/* It's not clear if this issue actually occurs in real
+		 * use-cases, so stick with the simple solution until the
+		 * first bug report.
+		 */
+		LOG_ERROR("BUG: Delay %s (%d) is not increased anymore.",
```

Comment:
Detail: I have also reworded the message a little. I would also prefer to not call this "BUG" since this can happen due to reasons outside of OpenOCD's developer control.

I would keep this as an error to make sure the message is very visible, should this ever happen.

```suggestion
		LOG_ERROR("Delay for %s (%d) is not increased anymore (maximum was reached).",
```

Reply:
Adressed.